README.md

```diff
@@ -110,37 +109,54 @@ Next run a suite of tests with `make test` and see a corresponding walkthrough
 of API features in [tests/demo/walkthrough.sh](tests/demo/walkthrough.sh). For
 more demonstrations, see the [demos](demos) directory.
 
-## Running the Registry API server locally
+## Running the Registry API server
+
+### Configuration
+
+Configuration for `registry-server` is loaded from the
+`$HOME/.config/registry/` directory. YAML, JSON, TOML, and other common formats
```

Comment:
"other common formats" is incredibly broad... is that phrase from the viper docs? I think we should link to them here so that people could see exactly which other common formats are supported.

Reply:
Good point, those are just the formats I remembered Viper supports. Their docs include a lot of irrelevant information, so I would prefer to simply list off file types that are supported. 

>Viper supports JSON, TOML, YAML, HCL, INI, envfile and Java Properties files

We can list all of those or just a subset of them so we don't have to worry about supporting so many types if for some reason we ever switched libraries (unlikely I think). It might be easier to explain if only support YAML, for example:

>Configuration for `registry-server` is loaded from `$HOME/.config/registry/registry-server.yaml`.

This reminds me, I added a config path check on the current working directory for `registry-server` to support copying the config file into docker containers. Do we want to continue copying our example configs into docker containers? It seems like a bad way of setting configuration compared to environment variables in that scenario.

Reply:
How about saying that we recommend YAML but use Viper which supports other formats (and link to Viper docs, with no further discussion)?

On configuration, I think there are situations where config yaml files might be preferred. Kubernetes uses YAML extensively, for example.

On this, I noticed that container builds might now have to copy their config files into a weird location (for a container) - $HOME/.config/registry/registry-server.yaml where presumably $HOME is the home directory of the user running inside the container (often root). It would be less confusing to directly point to a configuration file (perhaps with "--config", which [Hugo](https://gohugo.io/getting-started/usage/) uses). Could we add that (restore -c but with a possibly-different name)?

Reply:
>How about saying that we recommend YAML but use Viper which supports other formats (and link to Viper docs, with no further discussion)?

Good idea. Let's recommend YAML and refer generally to Viper's documentation for other possible formats.

>On this, I noticed that container builds might now have to copy their config files into a weird location

Yeah that's what I was pointing out, and it's why I added the current working directory as a location where config files can be read from. No weird location is needed.

>It would be less confusing to directly point to a configuration file (perhaps with "--config", which Hugo uses). Could we add that (restore -c but with a possibly-different name)?

The Dockerfile uses `COPY config/registry-server.yaml /registry-server.yaml`, which places the config in the same directory as the `registry-server` program where it will be automatically loaded.

Do you still think we should add the flag? It seems like a lot of work for the amount of benefit. We would have to add custom code to support the flag and explain the precedence of that config compared to other loading options.

Reply:
Ah, I see that's what you meant by "I added a config path check on the current working directory for registry-server"... is that mentioned in the README? (I missed it, if so). 

I'm guessing the order of precedence is:
1. environment variables
2. a registry-server.yaml in the current working directory
3. $HOME/.config/registry/registry-server.yaml

I think accidental confusion of 2 and 3 would make it better to have an explicit parameter (someone might run the registry-server without noticing that there's a config file in the current directory, or they might change directories and get an unexpected config).  An explicit --config would reduce this and would also let people name their config files whatever they want - for example, suppose someone wants (like we did) to have a directory with different config files for different setups. Then the precedence would probably be

1. environment variables
2. a file the user specifies with --config
3. $HOME/.config/registry/registry-server.yaml

I think following precedent is good here, and --config is used by Hugo (I didn't check other viper users, but Hugo seems fairly popular)

Reply:
I would like to have the `--config` flag included in this PR, and use it to simplify the precedence logic (above) so that we don't have to hide information from users to avoid confusion. (I don't think giving the working directory lower priority than `$HOME/.config/registry` is intuitive, so it's at least arguable, and using the flag removes the need to implicitly read a local config file.)

Here are three Viper users listed at https://github.com/spf13/viper that have `--config` options (there may be more, these were the first three I checked):
- https://github.com/digitalocean/doctl
- https://github.com/jgsqware/clairctl
- https://github.com/theupdateframework/notary (https://github.com/theupdateframework/notary/blob/bf96a202a09a312ae005cd312fc06ff4d2c166ce/cmd/notary-server/main.go#L37)

`gcloud` uses `--configuration` and `kubectl` uses `--kubeconfig` for similar options. I don't think we have a reason to deviate from the `--config` norm as they do, though. 
- `gcloud` configurations are a [complex topic](https://github.com/spf13/viper) that is more than a typical "config" file, and
-  Kubernetes uses `kubeconfig` to describe [client configurations](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/). This PR is for a server configuration.

Reply:
Done.